test/render.cpp

```diff
@@ -0,0 +1,71 @@
+#include "render.h"
+#include "HdrImage.h"
+#include "camera.h"
+#include "imagetracer.h"
+#include <cassert>
+
+using namespace std;
+
+void test_OnOff_render() {
+
+  Sphere sphere{
+      translation(Vec(2.0, 0.0, 0.0)) * scaling(Vec(0.2, 0.2, 0.2)),
+      Material(make_shared<DiffusiveBRDF>(make_shared<UniformPigment>(WHITE)))};
+  HdrImage image(3, 3);
+  OrthogonalCamera camera(1.);
+  ImageTracer tracer(image, make_shared<OrthogonalCamera>(camera));
+  World world;
+  world.add(make_shared<Sphere>(sphere));
+  OnOffRenderer renderer(world);
+  tracer.fire_all_rays(renderer);
+
+  assert(image.get_pixel(0, 0).is_close(BLACK));
+  assert(image.get_pixel(1, 0).is_close(BLACK));
+  assert(image.get_pixel(2, 0).is_close(BLACK));
+
+  assert(image.get_pixel(0, 1).is_close(BLACK));
+  cout << image.get_pixel(1, 1).r << " " << image.get_pixel(1, 1).g << " "
+       << image.get_pixel(1, 1).b << " " << endl;
+  assert(image.get_pixel(1, 1).is_close(WHITE));
```

Comment:
It seems that the ray does not hit the sphere created in world: so render return BLACK and not WHITE as we expect. The same problem appears at line 58 in `test_flat_render`.
The strange things is that the tests relative to `Shape` and `World` work.